media-libs/zxing-cpp/zxing-cpp-2.2.1-r1.ebuild

```diff
@@ -0,0 +1,95 @@
+# Copyright 1999-2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+PYTHON_COMPAT=( python3_{10..13} )
+
+DISTUTILS_EXT=1
+DISTUTILS_OPTIONAL=1
+DISTUTILS_USE_PEP517=setuptools
+
+inherit cmake distutils-r1
+
+DESCRIPTION="C++ Multi-format 1D/2D barcode image processing library"
+HOMEPAGE="https://github.com/zxing-cpp/zxing-cpp"
+SRC_URI="https://github.com/${PN}/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz
+	test? ( https://github.com/${PN}/${PN}/releases/download/v${PV}/test_samples.tar.gz -> ${P}-test.tar.gz )
+"
+
+LICENSE="Apache-2.0"
+SLOT="0/3"
+KEYWORDS="~amd64 ~arm ~arm64 ~loong ~ppc64 ~riscv ~x86"
+IUSE="python test"
+
+DEPEND="python? ( dev-python/pybind11 )"
+RDEPEND="${PYTHON_DEPS}"
+BDEPEND="${DISTUTILS_DEPS}
+	test? (
+		dev-cpp/gtest
+		dev-libs/stb
+		dev-libs/libfmt
+	)
+"
+REQUIRED_USE="${PYTHON_REQUIRED_USE}"
+RESTRICT="!test? ( test )"
+
+# https://github.com/zxing-cpp/zxing-cpp/issues/848
+CMAKE_WARN_UNUSED_CLI=no
+QA_PRESTRIPPED="usr/lib/python.*/site-packages/zxingcpp.cpython-.*.so"
```

Comment:
That looks like a bug to be fixed.

Reply:
I agree, which is why I left a link to the upstream TODO on line 37.

I sunk too many hours into attempting to resolve that without success, and checked with a few other developers in on Libera Chat. The consensus on fixing it was to "scream and bitterly weep" or overhaul huge portions of the upstream build process which is outside the scope of this PR.